snap/snapcraft.yaml

```diff
@@ -200,6 +194,10 @@ parts:
     override-build: |
         ${SNAP}/libexec/snapcraft/craftctl default
 
+        PYTHONPATH=$CRAFT_PART_INSTALL/lib/python3.12/site-packages \
```

Comment:
can we experiment with doing this completion work in a separate part? I think it'll be cleaner and simpler to maintain. Something like:

```yaml
completion-script:
  after: [snapcraft]
  plugin: nil
  build-environment:
    - PYTHONPATH: $CRAFT_STAGE/lib/python3.12/site-packages
  override-build: |
    python3 -m craft_cli.completion ...

```